spring-jdbc/src/main/java/org/springframework/jdbc/support/GeneratedKeyHolder.java

```diff
@@ -61,26 +62,34 @@ public GeneratedKeyHolder(List<Map<String, Object>> keyList) {
 	@Override
 	@Nullable
 	public Number getKey() throws InvalidDataAccessApiUsageException, DataRetrievalFailureException {
+		return getKeyAs(Number.class);
+	}
+
+	@Override
+	@Nullable
+	public <T> T getKeyAs(Class<T> keyClass) throws InvalidDataAccessApiUsageException, DataRetrievalFailureException {
 		if (this.keyList.isEmpty()) {
 			return null;
 		}
 		if (this.keyList.size() > 1 || this.keyList.get(0).size() > 1) {
 			throw new InvalidDataAccessApiUsageException(
 					"The getKey method should only be used when a single key is returned.  " +
-					"The current key entry contains multiple keys: " + this.keyList);
+							"The current key entry contains multiple keys: " + this.keyList);
 		}
 		Iterator<Object> keyIter = this.keyList.get(0).values().iterator();
 		if (keyIter.hasNext()) {
 			Object key = keyIter.next();
-			if (!(key instanceof Number)) {
+			if (key == null) {
+				return null;
+			}
+			if (!(keyClass.isAssignableFrom(key.getClass()))) {
 				throw new DataRetrievalFailureException(
 						"The generated key is not of a supported numeric type. " +
-						"Unable to cast [" + (key != null ? key.getClass().getName() : null) +
-						"] to [" + Number.class.getName() + "]");
+								"Unable to cast [" + key.getClass().getName() +
+								"] to [" + keyClass.getName() + "]");
 			}
-			return (Number) key;
-		}
-		else {
+			return keyClass.cast(key);
+		} else {
```

Comment:
In line with Spring's [code style](https://github.com/spring-projects/spring-framework/wiki/Code-Style), please ensure that `else` statements come on the next line.